TransportationSplitter.py

```diff
@@ -329,18 +395,65 @@ def get_trs(turn_restrictions, connector_ids):
 
     return trs_to_keep, flattened_tr_seq_items
 
-def get_split_segment_dict(original_segment_dict, split_segment, lr_columns_for_splitting):
+def get_destinations(destinations, connector_ids):
+    if destinations is None:
+        return None    
+    destinations_to_keep = [d for d in destinations if destination_applies_to_split_connectors(d, connector_ids)]     
+    return destinations_to_keep if destinations_to_keep else None
+
+def destination_applies_to_split_connectors(d, connector_ids):
+    if not connector_ids or len(connector_ids) != 2:
+        # at this point modified segments are expected to have exactly two connector ids, skip edge cases that don't
+        return False
+    when_heading = d.get("when", {}).get("heading")
+    from_connector_id = d.get("from_connector_id")
+    if not from_connector_id or not when_heading:
+        #these are required properties and need them to exist to resolve the split reference
+        return False
+    if when_heading == "forward" and connector_ids[1] != from_connector_id:
+        # the second connector id on this segment split needs to match the from_connector_id in the destination because heading scope applies only to forward
+        return False
+    if when_heading == "backward" and connector_ids[0] != from_connector_id:
+        # the first connector id on this segment split needs to match the from_connector_id in the destination because heading scope applies only to backward
+        return False
+    if from_connector_id not in connector_ids:
+        # the from_connector_id needs to be in this split's connector_ids no matter what
+        return False
+    return True
+
+def get_length_bucket(length):
```

Comment:
not sure which versions of python we support but this method could be an ideal candidate for a Structural Pattern Matching expression, https://stackoverflow.com/a/11479840 and https://peps.python.org/pep-0634/
it's only supported as of python `3.10`...

Reply:
as discussed, we'll probably want to avoid things that limit the versions this can run on